services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -12,22 +13,59 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
   def getProjectById(projectId: ProjectId): Future[Option[Project]] =
     projectRepository.getProjectById(projectId)
 
-  def addProject(request: ProjectAdditionRequest): Future[ProjectId] = {
+  def getProjectByName(namePattern: String, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectByName(namePattern)
+    result.flatMap {
+      case Some(project) if project.ownerId == userId => result
+      case Some(_)                                    => Future.failed(new ProjectAccessDeniedException)
+      case None                                       => Future.failed(new ProjectNotFoundException)
+    }
+  }
+
+  def addProject(request: ProjectAdditionRequest, userId: UserId): Future[ProjectId] = {
     val project =
       Project(
         projectId = ProjectId(UUID.randomUUID().toString),
-        ownerId = request.ownerId,
+        ownerId = userId,
         name = request.name,
-        repository = request.repository,
+        repository = "test_repo",
```

Comment:
`test_repo`?

Reply:
It is stub for the project addition operation request. User can create a project only in already existing repository. We discussed it with @mnesvit. May be i understood this incorrectly. Before it repository added from projectAdditionRequest.

Reply:
done